### PR TITLE
DATA-3444 Add exportTabularData to data client

### PR DESCRIPTION
**Pull request by katiepeters**
This PR adds the new Data endpoint [ExportTabularData](https://github.com/viamrobotics/app/pull/6969/files) to the typescript SDK.

### CODE REVIEW

File: src/app/data-client.ts [hunk 1]
```diff
@@ -38,6 +38,22 @@ interface TabularData {
   timeReceived?: Date;
 }
 
+interface TabularDataPoint {
+  partId: string;
+  resourceName: string;
+  resourceSubtype: string;
+  methodName: string;
+  timeCaptured: Date;
```
Comment (stuqdog) [inline src/app/data-client.ts:46]: per failing tests, I think this needs to be `Date | undefined` since it's possible no `timeCaptured` exists on the response.

**Review by stuqdog — APPROVED** [review]
The failing test needs to be addressed, otherwise lgtm!

File: src/app/data-client.ts [hunk 2]
```diff
@@ -53,6 +69,68 @@ export class DataClient {
...
+   * @returns An array of unified tabular data and metadata.
+   */
+  async exportTabularData(
+    partId: string,
+    resourceName: string,
+    resourceSubtype: string,
+    methodName: string,
+    startTime?: Date,
+    endTime?: Date
+  ) {
```
Comment (katiepeters) [inline src/app/data-client.ts:92]: @stuqdog Wondering if I should change this to return `AsyncIterableIterator<TabularDataPoint>`. Thoughts?

File: src/app/data-client.ts (hunk 1 above)
Comment (katiepeters) [reply src/app/data-client.ts:46]: I didn't realize we had to account for that with protobuf! Noted

File: src/app/data-client.ts (hunk 2 above)
Comment (stuqdog) [reply src/app/data-client.ts:92]: Hmm... I suspect we need to be `await`ing ourselves either way to convert into a `TabularDataPoint` (or get into some funky async functional mapping). I guess that might be useful if someone just wants to stream all tabular data as it continues to come in? That seems like it adds more complexity/work to users so my thought is no, but there may be a use case I'm not thinking about (could someone use that to tail all data as it comes in? is that a use case we expect to be common?).

tl;dr weakly I support keeping as-is but open to being convinced otherwise.

Comment (katiepeters) [reply src/app/data-client.ts:92]: I'm thinking of the use case where someone wants to retrieve a lot of data. Possibly even download it as a file
